File: TEST data for large repo/Repo Copy-3/Blueprism/BluePrism.Automate/AutomateAppCore/EnvironmentFunctions/IsStopRequestedFunction.vb
Imports BluePrism.AutomateProcessCore
Imports BluePrism.BPCoreLib.DependencyInjection

Namespace EnvironmentFunctions

    ''' <summary>
    ''' Function to check if a stop has been requested on the current session
    ''' </summary>
    ''' <remarks>This is usually the result of a "Safe Stop" request by a control rool user</remarks>
    Public Class IsStopRequestedFunction : Inherits IsStopRequestedFunctionBase

        Private ReadOnly mServerFactory As Func(Of IServer)

        Public Sub New(serverFactory As Func(Of IServer))
            mServerFactory = serverFactory
        End Sub

        Protected Overrides Function InnerEvaluate(parameters As IList(Of clsProcessValue), process As clsProcess) As clsProcessValue

            If IsNothing(process.Session) OrElse process.Session.Identifier.SessionIdentifierType <> SessionIdentifierType.RuntimeResource Then
                Return False
            End If

            Dim sessionId = CType(process.Session.Identifier, RuntimeResourceSessionIdentifier)

            Return mServerFactory().IsStopRequested(sessionId.SessionNumber)
        End Function
    End Class
End Namespace

File: TEST data for large repo/repo_Copy-2/Blueprism/BluePrism.DigitalWorker/EnvironmentFunctions/IsStopRequestedFunction.cs
using System;
using System.Collections.Generic;
using BluePrism.AutomateAppCore;
using BluePrism.AutomateProcessCore;
using BluePrism.DigitalWorker.Sessions;

namespace BluePrism.DigitalWorker.EnvironmentFunctions
{
    public class IsStopRequestedFunction : IsStopRequestedFunctionBase
    {
        private readonly IRunningSessionRegistry _runningSessionRegistry;

        public IsStopRequestedFunction(IRunningSessionRegistry runningSessionRegistry) 
        {
            _runningSessionRegistry = runningSessionRegistry ?? throw new ArgumentNullException(nameof(runningSessionRegistry));
        }

        protected override clsProcessValue InnerEvaluate(IList<clsProcessValue> parameters, clsProcess process)
        {
            if (process.Session == null)
                return false;

            var session = _runningSessionRegistry.Get(process.Session.ID);

            return session?.StopRequested ?? false;
        }
    }
}
